### PR TITLE
New Eslint rule: no shadow

### PR DESCRIPTION
**Pull request by kkoskelin**
Avoids declaring an inner-scope variable with same name as an outer-scope variable.  Handy for catching potential bugs during development.  Once we enable the `builtinGlobals: true` it will also prevent us from accidentally shadowing globals like 'document', 'window', 'body', and more.

### CODE REVIEW

File: web-client/src/views/CaseDetail/PetitionerInformation.jsx [hunk 1]
```diff
@@ -134,14 +134,12 @@ const PetitionerInformation = connect(
                   </h3>
                   <div>
                     <address aria-labelledby="secondary-label">
-                      {formattedCaseDetail.contactSecondary.name && (
```
Comment (rachelschneiderman) [inline web-client/src/views/CaseDetail/PetitionerInformation.jsx:137]: Was this line intentionally removed?

File: web-client/src/views/CreateOrder/TextEditor.jsx [hunk 2]
```diff
@@ -25,7 +25,6 @@ export const TextEditor = ({
   updateFormValueSequence,
   updateScreenMetadataSequence,
 }) => {
-  const inlineStylesFontSizes = {};
```
Comment (rachelschneiderman) [inline web-client/src/views/CreateOrder/TextEditor.jsx:28]: Should the reference to this on line 86 be removed?

File: web-client/src/views/CaseDetail/PetitionerInformation.jsx (hunk 1 above)
Comment (kkoskelin) [reply web-client/src/views/CaseDetail/PetitionerInformation.jsx:137]: Yes, this condition is actually being checked in the parent-block.  By the time we arrive here, we already know the condition to be true.

File: web-client/src/views/CreateOrder/TextEditor.jsx (hunk 2 above)
Comment (kkoskelin) [reply web-client/src/views/CreateOrder/TextEditor.jsx:28]: I believe the reference on line 86 will point to the `inlineStylesFontSizes` declared in the outer scope (near beginning of file)

**Review by rachelschneiderman — APPROVED** [review]

**Review by ericsorenson — APPROVED** [review]

**Pull request merged**